tools/indexer/pkg/http/client.go

```diff
@@ -0,0 +1,82 @@
+package http
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"net"
+	"net/http"
+	"strconv"
+	"time"
+)
+
+func NewHTTPClient(timeout time.Duration) *http.Client {
+	return &http.Client{
+		Timeout: timeout,
+		Transport: &http.Transport{
+			DialContext: (&net.Dialer{
+				Timeout:   5 * time.Second,
+				KeepAlive: 30 * time.Second,
+			}).DialContext,
+			MaxIdleConns:        100,
+			IdleConnTimeout:     90 * time.Second,
+			TLSHandshakeTimeout: 5 * time.Second,
+		},
+	}
+}
+func FetchJSONWithRetry(ctx context.Context, httpc *http.Client, url string, out any, attempts int, baseDelay time.Duration) error {
```

Comment:
If you want to provide these type of utility functions we cannot just assume this is doing the right thing. Unit tests should be added for this pkg. If not we can use some library that provides this functionality.